Added more documentation on solvepnp

neel-j-dev opened the pull request:
Added more information on what the camera matrix and distortion coefficients mean.
Added the point of calibrating and the different types of distortion
Added what the NT values for targetPose mean
Camera pitch and good practices for calibrating

neel-j-dev committed:
Test

neel-j-dev committed:
Added more info on solvepnp

neel-j-dev committed:
Type in features

modelmat commented on docs/source/advanced/solvePNP.rst:
```diff
@@ -5,14 +5,97 @@ SolvePNP
+   - distortion coefficents are vectors of up to 8 parameters, formated like : **k1, k2, p1, p2 [, k3 [, k4, k5, k6]]**
+   - **k1, k2..** are radical distortion coefficents
+   - **p1, p2...** are tangential distortion coefficents
+   - The distortion do not change based off the camera resolution, so the distortion coefficents will be the same for 640 x 480 images and 320 x 240 images
```
Does the `Camera Matrix` change based on camera resolution?

Also please indicate that this is an example, not an all-inclusive list.

neel-j-dev replied:
Yeah, camera matrices change proportionally to changes in resolution, added that in.

modelmat commented on docs/source/advanced/solvePNP.rst:
```diff
@@ -5,14 +5,97 @@ SolvePNP
+
+Now that we have callibrated the camera, and found the 3D world coordinates, we can now use this information to:
+
+- Undisort an image recieved from the camera
```
```suggestion
- Undistort an image received from the camera
```

modelmat commented on docs/source/advanced/solvePNP.rst:
```diff
@@ -5,14 +5,97 @@ SolvePNP
+it into the **Camera Pitch** field in the chameleon UI.
+
+.. image:: /images/solvepnp/pitch.PNG
+   :alt: Source: Research Gate
```
not exactly the point of alt text, and it's very vague as well.

neel-j-dev replied:
Alright took it out then

neel-j-dev committed:
Added camera matrix changes due to resolution changes

neel-j-dev committed:
Apply suggestions from code review

Co-Authored-By: Modelmat <Modelmat@users.noreply.github.com>

neel-j-dev committed:
Removed alt text

neel-j-dev committed:
Update solvepnp.rst

Clarified the radical and tangential are exaggerated for demonstrating and examples.

neel-j-dev committed:
Update solvepnp.rst

XzibitGG commented:
Yeah, camera matrices change proportionally to changes in resolution, added that in.

oriagranat9 commented on docs/source/advanced/solvePNP.rst:
```diff
@@ -5,14 +5,99 @@ SolvePNP
+   - **fx** and **fy** : These are the focal lengths of the camera expressed in pixels
+   - **cx** and **cy** : These are the principal points and refer to the image center 
+   - The camera matrix will be scaled proportionally to any change in resolution. For example, if resolution goes from 640 x 480 to 320 x 240 images
+   
```
you have to calibrate for each resolution we dont scale

neel-j-dev replied:
Want me to keep the explanation and write underneath that you'll need to recalibrate or scrap it and write it in place?

oriagranat9 replied:
except that everything is perfect i think you can just remove that sentence

neel-j-dev replied:
Alright, done

oriagranat9 commented on docs/source/advanced/solvePNP.rst:
```diff
@@ -5,14 +5,99 @@ SolvePNP
+   - distortion coefficents are vectors of up to 8 parameters, formated like : **k1, k2, p1, p2 [, k3 [, k4, k5, k6]]**
+   - **k1, k2..** are radical distortion coefficents
+   - **p1, p2...** are tangential distortion coefficents
+   - The distortion do not change based off the camera resolution, so the distortion coefficents will be the same for 640 x 480 images and 320 x 240 images
```
same here for calibration

neel-j-dev committed:
Update Solvepnp.rst

neel-j-dev committed:
Typo in solvepnp.rst

XzibitGG commented:
Doesn't look like the images got merged as well.
